rust/kernel/fs.rs

```diff
@@ -137,12 +137,138 @@ pub mod ro {
         }
     }
 
+    /// An inode that is locked and hasn't been initialised yet.
+    #[repr(transparent)]
+    pub struct NewINode<T: Type + ?Sized>(ARef<INode<T>>);
```

Comment:
no type invariant?

Reply:
below you write safety comments like:
```
// SAFETY: This is a new inode, so it's safe to manipulate it mutably.
// SAFETY: inode is a new inode, so it is valid for write.
// SAFETY: The new inode failed to be turned into an initialised inode, so it's safe
// (and in fact required) to call `iget_failed` on it.
```
and I do not particularly like them (but haven't given any thought as to what would fit better), maybe a type invariant might help here (e.g. the `ARef` is unique).